### DISCUSSION TITLE
Psych to codename chart converter?

### DISCUSSION THREAD

**Discussion by icepackisdripy** [CodenameCrew/CodenameEngine#448] [Q&A] [answered]
is there any way that i could port my psych engine chart into code name? I really don't want to chart a 7 minute song again.

**Comment by NexIsDumb** [comment, marked as answer]
dw codename automatically reads psych charts as long as theyre before of Psych 1.0
If theyre from Psych 1.0 just tell me cause we already had in mind to make the 1.0 converter

EDIT: the psych 1.0 charts reader was implemented in the public build already

**Reply by icepackisdripy** [reply]
yep the chart was made in 1.0

**Reply by NexIsDumb** [reply]
> yep the chart was made in 1.0

goddamn okay😭
i think its time to add 1.0 converter

**Reply by NexIsDumb** [reply]
pushed the commit for the 1.0 chart :p

**Reply by shadowlegion2377** [reply]
wait, so how to import the chart, do i just drag the (song-name).json chart into charts folder and name it normal?

**Reply by NexIsDumb** [reply]
> wait, so how to import the chart, do i just drag the (song-name).json chart into charts folder and name it normal?

in the file explorer, drag and drop the chart you want, inside songname/charts/ the json file name will be the name of the difficulty too unless you create a meta.json and mess with it

so yes what you said is right

**Reply by shadowlegion2377** [reply]
thanks, it's working now, i just forgot to delete the meta.json

**Reply by NexIsDumb** [reply]
ah i see

**Reply by Justamilkmangod** [reply]
hah, what about dadgum events bucko????

**Reply by r6915ee** [reply]
> hah, what about dadgum events bucko????

what

**Reply by Justamilkmangod** [reply]
> > hah, what about dadgum events bucko????
> 
> what

Like the events in the song doesn't port over, like flashes, scroll speed changes ect.

**Reply by r6915ee** [reply]
Codename has no functionality for porting over events, due to inconsistent event names, with the exception of camera switching. You will have to place those manually.

**Comment by vanillawdfpijawfo8iahjuw** [comment]
have you guys uploaded the psych engine converter? cant find it

**Reply by r6915ee** [reply]
If you're talking about the one for Psych 1.0, then I would maybe assume that it's already in the development repository, though I don't know how it works.

If you're talking about pre-1.0, then that's already in the main repository, thus it should be in your build. You just need to move or copy the chart into your song's `charts` directory and name it accordingly.

**Reply by NexIsDumb** [reply]
both are already in game, the public build too, like both 1.0 and psych legacy

**Reply by r6915ee** [reply]
Ah, alright then. That's good to know.
